src/Trainer.py

```diff
@@ -2,12 +2,13 @@
 import torch.nn as nn
 from torch.utils.data import DataLoader
 from torchvision import transforms
-from CLIP import CLIP
 from PIL import Image
-from torch.nn import Functional as F
+from torch.nn import functional as F
 from transformers import CLIPTokenizer as HFCLIPTokenizer
 import numpy as np
-def train_clip(model, device, dataset, learning_rate=1e-4, batch_size=16, warmup_steps=2000, num_epochs=50, enable_amp = True):
+import os
+
```

Comment:
Instead of loading the data inside the train function, I pass in the data loader itself. Added a couple of lines of code to save the checkpoints.